src/Symfony/Component/Message/MessageBus.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Message;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ * @author Matthias Noback <matthiasnoback@gmail.com>
+ */
+class MessageBus implements MessageBusInterface
+{
+    /**
+     * @var MiddlewareInterface[]
+     */
+    private $middlewares;
+
+    /**
+     * @param MiddlewareInterface[] $middlewares
+     */
+    public function __construct(array $middlewares = array())
+    {
+        $this->middlewares = $middlewares;
```

Comment:
to leverage `tagged` (which implies `iterable` type), what about;

```php
$this->middlewares  = is_array($middlewares) ? array_values($middlewares) : iterator_to_array($middlewares, false);
```

or find a lazy way?

Reply:
Yup, good point. Why would you `array_values` if array?

Reply:
to re-index middlewares, ensuring `callableForNextMiddleware` implem always works. (needs an `int $index` type btw :))